src/Dom/Document.php

```diff
@@ -1099,6 +1110,38 @@ private function convertAmpBindAttributes($html)
         return (null !== $converted) ? $converted : $html;
     }
 
+    /**
+     * Convert AMP bind-attributes back to their original syntax.
+     *
+     * This is not guaranteed to produce the exact same result as the initial markup, as it is more of a best guess.
+     * It can end up replacing the wrong attributes if the initial markup had inconsistent styling, mixing both syntaxes
+     * for the same attribute. In either case, it will always produce working markup, so this is not that big of a deal.
+     *
+     * @see convertAmpBindAttributes() Reciprocal function.
+     * @link https://www.ampproject.org/docs/reference/components/amp-bind
+     *
+     * @param string $html HTML with amp-bind attributes converted.
+     * @return string HTML with amp-bind attributes restored.
+     */
+    public function restoreAmpBindAttributes($html)
+    {
+        if (empty($this->convertedAmpBindAttributes)) {
+            return $html;
+        }
+
+        $pattern     = sprintf(
+            '#%s(%s)#i',
+            self::AMP_BIND_DATA_ATTR_PREFIX,
+            implode('|', array_unique($this->convertedAmpBindAttributes))
+        );
+        $replacement = '[$1]';
+        $limit       = count($this->convertedAmpBindAttributes);
+
+        $restored = preg_replace($pattern, $replacement, $html, $limit);
+
+        return (null !== $restored) ? $restored : $html;
+    }
```

Comment:
I still don't believe that _restoring_ is the right approach here since attributes can be added/removed after the DOM has been parsed and before it has been serialized, resulting in a incorrect restoration as the counts will change: https://github.com/ampproject/amp-toolbox-php/commit/bb607f85ae3e277d67627d03b9e0ea80f6ab2476#commitcomment-47328593

As mentioned in https://github.com/ampproject/amp-toolbox-php/commit/bb607f85ae3e277d67627d03b9e0ea80f6ab2476#commitcomment-47328668:

> Therefore, I think the only reasonable path is to introduce a `useBracketedAmpBindSyntax` flag which controls whether _all_ attributes are serialized using the bracketed syntax or the `data-amp-bind-` syntax. By default it would be `false`, but in unit tests you could make it `true`.

Reply:
So, what I did now is to provide a way to pass options around the document (with BC fallback that mimic the built-in `DOMDocument` do its syntax still works) and I added an option (`Document::OPTION_AMP_BIND_SYNTAX`) to configure how the conversion should be handled for `amp-bind` attributes. There are three available options:
- `Document::AMP_BIND_SYNTAX_AUTO` => try a best guess to keep the individual attributes in the syntax they were in originally.
- `Document::AMP_BIND_SYNTAX_DATA_ATTRIBUTE` => convert all `amp-bind` syntax into the `data-amp-bind-*` syntax.
- `Document::AMP_BIND_SYNTAX_SQUARE_BRACKETS` => convert all `amp-bind` attributes into the `[*]` syntax.

We can then set the option to `Document::AMP_BIND_SYNTAX_DATA_ATTRIBUTE` in the WordPress plugin to keep the current behavior for WP unchanged.

The default is `Document::AMP_BIND_SYNTAX_AUTO`, as I think this is the expected behavior, as third-party users of the library would not even be aware the `DOMDocument` code is broken and requires such a behind-the-scenes conversion in the first place.